src/Data/Bool/Compat.hs

```diff
@@ -12,6 +15,41 @@ module Data.Bool.Compat (
 import Data.Bool
 
 #if !MIN_VERSION_base(4,7,0)
+
+-- These instances are only valid if Bits isn't a subclass of Num (as Bool is
+-- not a Num instance), which is only true as of base-4.6.0.0 and later.
+# if MIN_VERSION_base(4,6,0)
+import Data.Bits (Bits(..))
+import Data.Eq ((/=))
+
+instance Bits Bool where
```

Comment:
Sure, works for me. It also hit me that there are some orphan instances that are available in one module and not the other (e.g., there's a `Generic (Const a b)` instance in `GHC.Generics.Compat` but not in `Control.Applicative.Compat`), so I'll fix that too.